src/librustc_typeck/collect.rs

```diff
@@ -1794,7 +1795,7 @@ fn explicit_predicates_of<'a, 'tcx>(
     // on a trait we need to add in the supertrait bounds and bounds found on
     // associated types.
     if let Some((_trait_ref, _)) = is_trait {
-        predicates.extend(tcx.super_predicates_of(def_id).predicates);
+        predicates.extend(tcx.super_predicates_of(def_id).predicates.iter().map(|p| *p));
```

Comment:
I think the `.map(|p| *p)` here and elsewhere could be replaced by `.cloned()`.

Reply:
Won't that cause additional refcounting that the `.map(|p| *p)` won't?

Reply:
No, it does `.map(|p| p.clone())` which is the same as `.map(|p| *p)` when the latter works because the latter only works when `typeof(*p): Copy` and `Copy` implies `Clone` is just a copy.